radixdlt-java/src/main/java/com/radixdlt/client/core/network/reducers/RadixNetwork.java

```diff
@@ -0,0 +1,53 @@
+package com.radixdlt.client.core.network.reducers;
+
+import com.radixdlt.client.core.network.RadixNetworkState;
+import com.radixdlt.client.core.network.RadixNode;
+import com.radixdlt.client.core.network.RadixNodeAction;
+import com.radixdlt.client.core.network.RadixNodeState;
+import com.radixdlt.client.core.network.actions.AddNodeAction;
+import com.radixdlt.client.core.network.actions.GetNodeDataResultAction;
+import com.radixdlt.client.core.network.actions.WebSocketEvent;
+import com.radixdlt.client.core.network.websocket.WebSocketStatus;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Reducer which controls state transitions as new network actions occur.
+ * Explicitly does not contain state and should be maintained as a pure function.
+ */
+public final class RadixNetwork {
+	public RadixNetwork() {
+	}
+
+	public RadixNetworkState reduce(RadixNetworkState state, RadixNodeAction action) {
+		Map<RadixNode, RadixNodeState> newMap = null;
+		if (action instanceof GetNodeDataResultAction) {
```

Comment:
Oh this is really cool. 
I think we should consider splitting this up into multiple methods, as all the `if (action instanceof ...)` seem to be independent and repetitive.

Reply:
I actually think keeping it all in one method is cleaner as each transition is a small amount of logic and you need to do a lot less code jumping. Could perhaps refactor later if it gets too unwieldy.